### DISCUSSION TITLE
Hot reload support

### DISCUSSION THREAD

**Discussion by Creekmind** [LibraStack/UnityMvvmToolkit#68] [Q&A] [answered]
Hi!

I have created demo project with one View and ViewModel. Set default value for BindableLabel in ViewModel and UXML. Started project and change value of bound property in ViewModel. After hot reloading state has been gone.

<details><summary>View.cs</summary>
<p>

```cs
using UnityMvvmToolkit.UITK;

namespace HUD
{
    public class View : DocumentView<MainViewModel>
    {
    }
}

```

</p>
</details>

<details><summary>ViewModel.cs</summary>
<p>

```cs
namespace HUD
{
    public class MainViewModel : IBindingContext
    {
        [Observable] public Property<string> Value = new("Bind value");

        public MainViewModel()
        {
            Debug.Log("creating MainViewModel");
        }
    }
}

```

</p>
</details>

<details><summary>View.uxml</summary>
<p>

```xml
<ui:UXML xmlns:ui="UnityEngine.UIElements"
         xmlns:uie="UnityEditor.UIElements"
         xsi="http://www.w3.org/2001/XMLSchema-instance"
         engine="UnityEngine.UIElements"
         editor="UnityEditor.UIElements"
         noNamespaceSchemaLocation="../../UIElementsSchema/UIElements.xsd"
         editor-extension-mode="False">

  <UnityMvvmToolkit.UITK.BindableUIElements.BindableLabel binding-text-path="Value"
                                                          text="Default value"
                                                          style="font-size: 64px;"/>

</ui:UXML>

```

</p>
</details>

Demo:

![Запись экрана 2024-01-13 в 10 56 46](https://github.com/LibraStack/UnityMvvmToolkit/assets/3481215/be787f9d-7c17-4d8c-9e1e-751c8f973050)

I tried sample Mvvm.Counter and there is same behavior. **But** in my general project there is one view, that doesn't loose its state after hot reload. So my question is: is hot reload supported or it is issue?

**Comment by ChebanovDD** [comment]
Hi there,

I don’t quite understand what you mean by hot reload?

Can you describe step by step what actions you perform and what result you expect?

P.S. What version of Unity do you use?

**Reply by Creekmind** [reply]
I'm using Unity 2023.1.14f1. By hot reload I mean applying changes to the project without stopping and starting it. 

My actions (also I have attached GIF below):
1. Create UXML with components bound to the view models with default values
2. Start project, all components renderers properly with values from view models
3. Make changes in code, e.g. change style of bindable component in UXML: change margin-left from 16px to 32px
4. Changes applied to project without restarting it, but all components unbound from view models and lost its values; see GIF below: 
    * Bind value -> Default value
    * Icon disappears
5. Make changes in code again, just add regular Label to UXML
6. Changes applied to project without restarting it: the label appears

Expected behavior: 
After any changes in code bindable components shouldn't loose their bound values

<details>
  <summary>GIF</summary>
  
![output (1)](https://github.com/LibraStack/UnityMvvmToolkit/assets/3481215/6684c375-2880-42fb-9a93-22f4a65362ed)

</details>

**Reply by ChebanovDD** [reply, marked as answer]
Unfortunately, this use case is not supported. But I’ll think about how to implement hot reload support in the feature.

**Reply by Creekmind** [reply]
Ok, I got it. Thanks. Anyway the lib is great 👍
